orttraining/orttraining/python/orttraining_pybind_state.cc

```diff
@@ -841,23 +841,19 @@ void addObjectMethodsForTraining(py::module& m, ExecutionProviderRegistrationFn
                                                                         non_trainable_tensor_protos, checkpoint_path));
         });
   m.def("load_checkpoint",
-        [](const std::string& checkpoint_path) {
+        [](const std::string& checkpoint_path, py::bytes& serialized_model) {
           std::vector<TensorProto> tensor_protos;
-          ORT_THROW_IF_ERROR(onnxruntime::training::api::LoadCheckpoint(checkpoint_path, tensor_protos));
-          std::vector<py::bytes> tensor_protos_pybytes(tensor_protos.size());
-
-          const auto parse_tensor_proto_to_pybytes =
-              [](std::vector<py::bytes>& tensor_protos_pybytes, const std::vector<TensorProto>& tensor_protos) {
-                for (size_t i = 0; i < tensor_protos.size(); ++i) {
-                  std::string tensor_proto_str;
-                  tensor_protos[i].SerializeToString(&tensor_proto_str);
-                  tensor_protos_pybytes[i] = tensor_proto_str;
-                }
-              };
+          ONNX_NAMESPACE::ModelProto model_proto;
+          std::string model_proto_str;
```

Comment:
Try to instantiate the variable close to where it is being used. Move it to just before `model_proto.SerializeToString(&model_proto_str);`